tests/cargo-add.rs

```diff
@@ -580,13 +598,89 @@ path = "dummy.rs"
         )
         .expect("Manifest is writeable");
 
-    Command::cargo_bin("cargo-add")
+    let assert = Command::cargo_bin("cargo-add")
+        .expect("can find bin")
+        .args(&["add", "../dependency"])
+        .env("CARGO_IS_TEST", "1")
+        .current_dir(manifest_path.path().parent().expect("there is a parent"))
+        .assert()
+        .success();
+    println!("Succeeded: {}", assert);
+    manifest_path.assert(
+        r#"[package]
+name = "cargo-list-test-fixture"
+version = "0.0.0"
+
+[lib]
+path = "dummy.rs"
+
+[dependencies]
+cargo-list-test-fixture-dependency = { version = "0.0.0", path = "../dependency" }
+"#,
+    );
+
+    // check this works with other flags (e.g. --dev) as well
+    let assert = Command::cargo_bin("cargo-add")
+        .expect("can find bin")
+        .args(&["add", "../dependency", "--dev"])
+        .env("CARGO_IS_TEST", "1")
+        .current_dir(manifest_path.path().parent().expect("there is a parent"))
+        .assert()
+        .success();
+    println!("Succeeded: {}", assert);
+    manifest_path.assert(
+        r#"[package]
+name = "cargo-list-test-fixture"
+version = "0.0.0"
+
+[lib]
+path = "dummy.rs"
+
+[dependencies]
+cargo-list-test-fixture-dependency = { version = "0.0.0", path = "../dependency" }
+
+[dev-dependencies]
+cargo-list-test-fixture-dependency = { path = "../dependency" }
```

Comment:
this wasn't introduced in the PR, but shouldn't we skip adding to dev-deps if it's present in deps? I guess same applies to build-deps.

Reply:
dev-dependencies can introduce features, make optional dependencies, required, etc.  We'd need to interrogate these different settings to make sure we weren't skipping something that needed to happen.